client/src/app/shared/validators/custom-validators.ts

```diff
@@ -0,0 +1,6 @@
+import { ValidatorFn, FormGroup, ValidationErrors } from '@angular/forms';
+
+export const durationValidator: ValidatorFn = (control: FormGroup): ValidationErrors | null => {
+    const regExp = /^\s*([0-9]+)(:)?([0-5][0-9]?)?\s*[h|m]?$/g;
```

Comment:
Comments... Pleae describe what this regex does (e.g. take it into parts and describe them, or do some examples)